velox/vector/FlatVector.h

```diff
@@ -107,7 +107,11 @@ class FlatVector final : public SimpleVector<T> {
         rawValues_(values_.get() ? const_cast<T*>(values_->as<T>()) : nullptr) {
     setStringBuffers(std::move(stringBuffers));
     VELOX_DCHECK_GE(stringBuffers_.size(), stringBufferSet_.size());
-
+    if constexpr (std::is_same<T, UnscaledShortDecimal>::value) {
+      VELOX_CHECK(type->isShortDecimal());
```

Comment:
Consider to use DCHECK here?
```
VELOX_DCHECK_EQ(std::is_same_v<xxx>, type->isShortDecimal());
VELOX_DCHECK_EQ(std::is_same_v<yyy>, type->isLongDecimal());
```
Also please use std::is_same_v as to be consistent with other part of code (there are some ongoing efforts to refactoring this)